### ISSUE TITLE
Command parsing is almost unusably strict

### ISSUE THREAD

**Issue by pbrisbin** [stackbuilders/dotenv-hs#141]
:wave: Hello there from one [`dotenv` library](https://hackage.haskell.org/package/load-env) author to another :)

I've been looking to covert away from my project to yours because you support the `$( )` syntax, and I thought that would be a great way to read secrets from AWS SSM during development.

I tried,

```sh
ASANA_API_KEY=$(aws --profile freckle-dev ssm get-parameter \
  --name /dev/asana-api-access-key \
  --query 'Parameter.Value' \
  --output text)
```

But got,

```
bugsnag-update: .env:2:20:
  |
2 | ASANA_API_KEY=$(aws --profile freckle-dev ssm get-parameter \
  |                    ^
unexpected space
expecting ')' or alphanumeric character
```

OK, that's fair. Maybe you don't want to get into shell words parsing. How about I stash this in a local script and call it as a single word:

```sh
ASANA_API_KEY=$(bin/get-asana-api-key)
```

```
bugsnag-update: .env:2:20:
  |
2 | ASANA_API_KEY=$(bin/get-asana-api-key)
  |                    ^
unexpected '/'
expecting ')' or alphanumeric character
```

Alright... Maybe you want to avoid some kind exploit by only allowing commands on `$PATH`?

No worries,

```sh
ASANA_API_KEY=$(get-asana-api-key)
```

```
PATH=./bin:$PATH stack run
bugsnag-update: .env:2:20:
  |
2 | ASANA_API_KEY=$(get-asana-api-key)
  |                    ^
unexpected '-'
expecting ')' or alphanumeric character
```

I give up, and here I am :)

**So my question**: are these limitations intentional (e.g. for security reasons), or would you accept a PR to relax them?

Ideally, I think you'd accept everything until the final (balanced) `)`, break it up with a library like [`shellwords`](https://hackage.haskell.org/package/shellwords), and then `exec` it. I think this would be safe, provided you always `exec` a command and arguments, and never involve a shell.

At a minimum, I'd like to see `/`, `-`, `.`, and `_` accepted.

**Comment by CristhianMotoche** [comment]
Hello @pbrisbin It seems we're parsing [alpha numeric characters](https://github.com/stackbuilders/dotenv-hs/blob/master/src/Configuration/Dotenv/Parse.hs#L78-L83) right now. I think we can relax that to allow the characters that you mention. So, feel free to open a PR. Thanks!

**Comment by pbrisbin** [comment]
Cool, can do. Would you actively be against me making `VAR=$(cmd arg arg)` work? I would find that incredibly useful, and don't mind implementing it. Since I author `shellwords`, I could even export its parser to be incorporated into yours.

**Comment by CristhianMotoche** [comment]
Hey @pbrisbin I won't mind if you import that library or the parser. It seems it's a light library, so it shouldn't increase the build time or the final dotenv binary.

**Comment by CristhianMotoche** [comment]
Solved by #142 and released in version 9.0.3. Thanks @pbrisbin
